Libraries/LibWeb/DOM/XMLHttpRequest.cpp

```diff
@@ -46,6 +46,12 @@ XMLHttpRequest::~XMLHttpRequest()
 {
 }
 
+void XMLHttpRequest::set_ready_state(ReadyState ready_state)
+{
+    // FIXME: call onreadystatechange once we have that
+    m_ready_state = ready_state;
+}
```

Comment:
A setter function for this might seem overkill _right now_ (as we could just `m_ready_state = ...`) but considering we need to call a callback in the future this made sense to me.

Reply:
That's fine :)